test/problematic_tests/impl_test.py

```diff
@@ -0,0 +1,48 @@
+import os
+import pytest
+import unittest
+
+from GenomeFileUtil.GenomeFileUtilImpl import GenomeFileUtil
+from conftest import assert_exception_correct
+
+
+class ImplTest(unittest.TestCase):
+    def setUp(self):
+        self.cfg = {}
+        self.cfg["workspace-url"] = "https://ci.kbase.us/services/ws"
+        self.cfg["shock-url"] = "https://ci.kbase.us/services/shock-api"
+        self.cfg["handle-service-url"] = "https://ci.kbase.us/services/handle_service"
+        self.cfg["SDK_CALLBACK_URL"] = "http://ip:port"
+        self.cfg["scratch"] = "/kb/module/work/tmp"
+        self.cfg["srv-wiz-url"] = "https://ci.kbase.us/services/service_wizard"
+        self.cfg["auth-service-url"] = "https://ci.kbase.us/services/auth/api/legacy/KBase/Sessions/Login"
+        self.cfg["re-api-url"] = "https://ci.kbase.us/services/relation_engine_api"
+        self.cfg["KB_AUTH_TOKEN"] = os.environ.get("KB_AUTH_TOKEN", None)
```

Comment:
The token isn't passed in the `GenomeFileUtil.__init__` config dict

Also, pulling the token from os.environ is bad practice (although again preceding you); the token should be pulled from the `ctx` object provided to the methods rather than the environment. In apps it winds up working but it's a bad habit as in dynamic services it'll either fail outright or screw up permissions

```suggestion
```

Reply:
updated!

Reply:
:+1: